mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/mapping/definition/annotation/VectorField.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.pojo.mapping.definition.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.hibernate.search.engine.backend.types.Projectable;
+import org.hibernate.search.engine.backend.types.Searchable;
+import org.hibernate.search.engine.backend.types.VectorSimilarity;
+import org.hibernate.search.engine.environment.bean.BeanRetrieval;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.processing.PropertyMapping;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.processing.PropertyMappingAnnotationProcessorRef;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.processing.impl.VectorFieldAnnotationProcessor;
+
+/**
+ * Maps an entity property to a vector field in the index.
+ * <p>
+ * This annotation will work for any {@code float/byte} array/list.
+ * <p>
+ * TODO: vector : docs
+ */
+@Documented
+@Target({ ElementType.METHOD, ElementType.FIELD })
+@Retention(RetentionPolicy.RUNTIME)
+@Repeatable(VectorField.List.class)
+@PropertyMapping(processor = @PropertyMappingAnnotationProcessorRef(type = VectorFieldAnnotationProcessor.class,
+		retrieval = BeanRetrieval.CONSTRUCTOR))
+public @interface VectorField {
+
+	/**
+	 * @return The name of the index field.
+	 */
+	String name() default "";
+
+	/**
+	 * @return Whether this field should be projectable.
+	 * @see Searchable
+	 */
+	Projectable projectable() default Projectable.DEFAULT;
+
+	int dimension();
+
+	/**
+	 * @return How vector similarity is calculated.
+	 * @see VectorSimilarity
+	 */
+	VectorSimilarity vectorSimilarity() default VectorSimilarity.L2;
+
+	/**
+	 * Open Search has more parameters ...
+	 * https://opensearch.org/docs/latest/search-plugins/knn/knn-index/#hnsw-parameters-1
+	 */
+
+	/**
+	 * @return A value used instead of null values when indexing.
+	 */
+	String indexNullAs() default AnnotationDefaultValues.DO_NOT_INDEX_NULL;
+
+	/**
+	 * The number of candidates to track while assembling the list of nearest neighbors for each new node. Defaults to 100 in Elasticsearch.
+	 * The size of the dynamic list used during k-NN graph creation. Higher values lead to a more accurate graph but slower indexing speed. Defaults to 512 in Open Search.
+	 * @return
```

Comment:
You might as well put `@return` at the start of the javadoc, like in other attributes?

Same below.